exporter/tanzuobservabilityexporter/metrics.go

```diff
@@ -345,3 +350,195 @@ func (s *sumConsumer) pushNumberDataPoint(
 		*errs = append(*errs, err)
 	}
 }
+
+// histogramReporting takes care of logging and internal metrics for histograms
+type histogramReporting struct {
+	logger                   *zap.Logger
+	malformedHistograms      counter
+	leInUse                  counter
+	noAggregationTemporality counter
```

Comment:
I know it may not feel "natural" for you to use other metrics library, but I would like in the future to change this to use same library as the rest of the collector code :) For the moment please create an issue to track this.

Reply:
Thank you for the feedback. I am a big fan of code reuse rather than reinventing the wheel. Do tell me where to find this library you speak of, so that I may use it instead of my own `counter` type.

Reply:
I created issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6819 for this comment.